invertedai/simulation.py

```diff
@@ -0,0 +1,106 @@
+from typing import Optional, List
+
+from invertedai.api_resources import initialize, drive
+
+
+class Simulation:
+    """
+    Stateful wrapper around Inverted AI API to simplify co-simulation.
+    Typically, it's sufficient to call `ego_agent_states` and `step`.
+    """
+    def __init__(
+            self,
+            # all arguments to initialize are also given to this constructor
+            location: str = "CARLA:Town03:Roundabout",
+            agent_count: int = 1,
+            # some further configuration options
+            monitor_infractions: bool = False,
+            render_birdview: bool = False,
+            ego_agent_mask: Optional[List[bool]] = None,
```

Comment:
Instead of using mask with for loops, we can  just use ego_agents_indices and do something like map(self._agent_states.__getitem__, self._ego_agents_indices) to mutate the agent_states directly

Reply:
I guess, but what would be the advantage?

Reply:
We're limited to 100 agents anyway, so I don't think we need to worry about that. I'd rather do what's easier to read.